src/models/user.jsx

```diff
@@ -0,0 +1,90 @@
+import {
+  collection,
+  doc,
+  getCountFromServer,
+  getDocs,
+  query,
+  // eslint-disable-next-line no-unused-vars
+  runTransaction,
+} from "firebase/firestore";
+import { createContext, useMemo } from "react";
+
+import database from "@/clients/firebase";
+import { JOB_STATUS } from "@/constants";
+
+// All just a copu of job.jsx more or less
+
+const UserContext = createContext({
+  approveUser: () => {},
+  updateUser: () => {},
+  deleteUser: () => {},
+  listUsers: (_queryConstraints) => [],
+  countUsers: (_queryConstraints) => Number,
+});
+
+const updateUser = async (userId, payload) => {
+  if (payload.id) {
+    delete payload.id;
+  }
+
+  await runTransaction(database, async (transaction) => {
+    const userDocRef = doc(database, "Users", userId);
+    const userDoc = await transaction.get(userDocRef);
+    if (!userDoc.exists()) {
+      throw `User ${userId} does not exist`;
+    }
+    transaction.update(userDocRef, payload);
+  });
+};
+
+const approveUser = async (userId) => {
+  await updateUser(userId, { status: JOB_STATUS.APPROVED });
```

Comment:
I guess we need some new constants for user status

Reply:
ok will do